src/signal_processing_algorithms/energy_statistics.py

```diff
@@ -0,0 +1,309 @@
+"""Energy statistics."""
+from dataclasses import dataclass
+from typing import List, Tuple, Union
+
+import numpy as np
+
+from signal_processing_algorithms.distance import get_distance_matrix
+
+
+@dataclass
+class EnergyStatisticsTestResult:
```

Comment:
I guess technically this is `EnergyStatistics` and the other is `EnergyStatisticsCalculator` or something like that.

Reply:
Made these two classes instead since EnergyCalculator should ideally not have e, t, h as its attributes and the pvalues are not exactly energy statistics but results of permutation tests.
  ```
 class EnergyStatistics:
    e: float
    t: float
    h: float
```

  ```
class EnergyStatisticsWithProbabilities(EnergyStatistics):
    e_pvalue: float
    t_pvalue: float
    h_pvalue: float
```